From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Stratum MC <build@stratum.mc>
Date: Fri, 30 May 2026 00:00:00 +0000
Subject: [PATCH] Stratum-server-hooks

diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 6a5d404e570ede5d49c82bccc30aa75ddd58983b..12998638e9591b2c742dec000b3affa8fa0936d9 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -333,6 +333,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         }
 
         // CraftBukkit start
+        mc.stratum.StratumPluginInstaller.installOrUpdate(); // Stratum
         this.server.loadPlugins();
         this.server.enablePlugins(org.bukkit.plugin.PluginLoadOrder.STARTUP);
         // CraftBukkit end
@@ -376,6 +377,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         this.services.nameToIdCache().resolveOfflineUsers(!this.usesAuthentication());
         LOGGER.info("Preparing level \"{}\"", this.getLevelIdName());
         this.loadLevel(this.storageSource.getLevelId()); // CraftBukkit
+        mc.stratum.StratumServer.getInstance().init(this); // Stratum
         long elapsed = Util.getNanos() - levelNanoTime;
         String time = String.format(Locale.ROOT, "%.3fs", elapsed / 1.0E9);
         LOGGER.info("Done preparing level \"{}\" ({})", this.getLevelIdName(), time); // Paper - Improve startup message, add total time

diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index cc35c6271c98615a486422d0490145227626bd85..efce534e1707cd3cd75b9ca38ccb8647f04ad256 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1046,6 +1046,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
         }
 
         LOGGER.info("Stopping server");
+        mc.stratum.StratumServer.getInstance().shutdown(); // Stratum
         Commands.COMMAND_SENDING_POOL.shutdownNow(); // Paper - Perf: Async command map building; Shutdown and don't bother finishing
         // CraftBukkit start
         if (this.server != null) {
