From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Stratum MC <build@stratum.mc>
Date: Fri, 30 May 2026 00:00:00 +0000
Subject: [PATCH] Stratum-server-hooks


diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index f008e09da3c2210a0d9a6ccae4d34f959d5ae5d0..681fe56a56bad5dde1c34470a4a7f0ea86e0db71 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -333,6 +333,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
 
         // CraftBukkit start
+        mc.stratum.StratumPluginInstaller.installOrUpdate(); // Stratum
         this.server.loadPlugins();
         this.server.enablePlugins(org.bukkit.plugin.PluginLoadOrder.STARTUP);
         // CraftBukkit end
@@ -376,6 +377,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         LOGGER.info("Preparing level \"{}\"", this.getLevelIdName());
         this.loadLevel(this.storageSource.getLevelId()); // CraftBukkit
+        mc.stratum.StratumServer.getInstance().init(this); // Stratum
         long elapsed = Util.getNanos() - levelNanoTime;
         String time = String.format(Locale.ROOT, "%.3fs", elapsed / 1.0E9);
diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index f18001d368d74b987b21521ac2ffe75479ba3a87..843bdad7c0141f0378eeeb0af0765abd1621572e 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -947,6 +947,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<Runnab
 
         LOGGER.info("Stopping server");
+        mc.stratum.StratumServer.getInstance().shutdown(); // Stratum
         Commands.COMMAND_SENDING_POOL.shutdownNow(); // Paper - Perf: Async command map building; Shutdown and don't bother finishing
         // CraftBukkit start
         if (this.server != null) {
